src/components/KInput/KInput.vue

```diff
@@ -319,196 +238,138 @@ export default {
+  }
 
-        :deep(svg) {
-          height: $kInputSmallIconSize;
-          width: $kInputSmallIconSize;
+  // slots styles
+  .input-element-wrapper {
+    position: relative;
+
+    .before-content-wrapper, .after-content-wrapper {
+      position: absolute;
+      top: 50%;
+      transform: translateY(-50%);
+
+      :deep(>) {
+        &:is(.kui-icon) {
+          color: var(--kui-color-text-neutral, $kui-color-text-neutral) !important;
+          /* stylelint-disable-next-line @kong/design-tokens/use-proper-token */
+          height: $kInputIconSize;
+          /* stylelint-disable-next-line @kong/design-tokens/use-proper-token */
+          width: $kInputIconSize;
         }
-      }
 
-      &.icon-start {
-        padding-left: calc($kInputSmallSizingX + var(--kui-space-40, $kui-space-40) + $kInputSmallIconSize) !important; // account for icon offset and width
-        ~ .input-icon {
-          left: $kInputSmallSizingX;
-        }
-      }
-      &.icon-end {
-        padding-right: calc($kInputSmallSizingX + var(--kui-space-40, $kui-space-40) + $kInputSmallIconSize) !important; // account for icon offset and width
-        ~ .input-icon {
-          right: $kInputSmallSizingX;
+        &:not(.kui-icon) {
+          display: none;
         }
       }
     }
 
-    // input size large
-    $kInputLargeSizingY: var(--kui-space-60, $kui-space-60);
-    $kInputLargeSizingX: var(--kui-space-80, $kui-space-80);
-    $kInputLargeIconSize: var(--kui-icon-size-60, $kui-icon-size-60);
+    .before-content-wrapper {
+      left: 0;
+      margin-left: $kInputPaddingX;
+    }
 
-    &.k-input-large {
-      ~ .input-icon {
-        // (height of entire element (paddings + line height) - icon size) / 2
-        top: calc((($kInputLargeSizingY + $kInputLargeSizingY + $kInputLineHeight) - $kInputLargeIconSize) / 2);
+    .after-content-wrapper {
+      margin-right: $kInputPaddingX;
+      right: 0;
+    }
 
-        :deep(svg) {
-          height: $kInputLargeIconSize;
-          width: $kInputLargeIconSize;
-        }
+    &.has-before-content {
+      .k-input {
+        padding-left: calc($kInputPaddingX * 2 + $kInputIconSize);
       }
+    }
 
-      &.icon-start {
-        padding-left: calc($kInputLargeSizingX + var(--kui-space-40, $kui-space-40) + $kInputLargeIconSize) !important; // account for icon offset and width
-        ~ .input-icon {
-          left: $kInputLargeSizingX;
-        }
-      }
-      &.icon-end {
-        padding-right: calc($kInputLargeSizingX + var(--kui-space-40, $kui-space-40) + $kInputLargeIconSize) !important; // account for icon offset and width
-        ~ .input-icon {
-          right: $kInputLargeSizingX;
-        }
+    &.has-after-content {
+      .k-input {
+        padding-right: calc($kInputPaddingX * 2 + $kInputIconSize);
       }
     }
   }
 }
 
-.help {
+.k-input {
+  background-color: var(--kui-color-background, $kui-color-background);
+  border: 0;
```

Comment:
issue: the border should be mapped to a token

Reply:
No because we really mean to remove border here. If we use `$kui-border-width-0` and then someone defines their custom value for that token, we will end up having both `border` and `box-shadow` on that element.